plugins/vault/vault.plugin.zsh

```diff
@@ -0,0 +1,408 @@
+        ': ::' \
+        ': ::' && ret=0
+}
+
+_vault_read() {
+    _arguments : \
+        ${general_args[@]} \
+        '(-format)-format=-[The format for output. By default it is a whitespace-delimited table. This can also be json.]:format:(json table)' \
+        '(-field)-field=-[If included, the raw value of the specified field will be output raw to stdout.]:field:' \
+        ': ::' && ret=0
+}
+
+_vault_commands() {
+    local -a commands
+
+    commands=(
+        "delete":"Delete operation on secrets in Vault"
+        "path-help":"Look up the help for a path"
+        "read":"Read data or secrets from Vault"
+        "renew":"Renew the lease of a secret"
+        "revoke":"Revoke a secret"
+        "server":"Start a Vault server"
+        "status":"Outputs status of whether Vault is sealed and if HA mode is enabled"
+        "write":"Write secrets or configuration into Vault"
+        "audit-disable":"Disable an audit backend"
+        "audit-enable":"Enable an audit backend"
+        "audit-list":"Lists enabled audit backends in Vault"
+        "auth":"Prints information about how to authenticate with Vault"
+        "auth-disable":"Disable an auth provider"
+        "auth-enable":"Enable a new auth provider"
+        "init":"Initialize a new Vault server"
+        "key-status":"Provides information about the active encryption key"
+        "mount":"Mount a logical backend"
+        "mount-tune":"Tune mount configuration parameters"
+        "mounts":"Lists mounted backends in Vault"
+        "policies":"List the policies on the server"
+        "policy-delete":"Delete a policy from the server"
+        "policy-write":"Write a policy to the server"
+        "rekey":"Rekeys Vault to generate new unseal keys"
+        "remount":"Remount a secret backend to a new path"
+        "rotate":"Rotates the backend encryption key used to persist data"
+        "seal":"Seals the vault server"
+        "ssh":"Initiate a SSH session"
+        "token-create":"Create a new auth token"
+        "token-renew":"Renew an auth token if there is an associated lease"
+        "token-revoke":"Revoke one or more auth tokens"
+        "unmount":"Unmount a secret backend"
+        "unseal":"Unseals the vault server"
+        "version":"Prints the Vault version"
+    )
+
+    _describe -t commands 'vault command' commands && ret=0
+}
+
+_vault() {
+    local curcontext=$curcontext ret=1
+
+    _arguments : \
+        ${main_args[@]} \
+        '*:: :->subcommands' && ret=0
+
+    if ((CURRENT == 1 )); then
+        _vault_commands && ret=0
+    fi
+
+    if [[ $state == subcommands ]]; then
+        # (( CURRENT -- ))
+        curcontext="${curcontext%:*:*}:vault-$words[1]:"
+        _call_function ret _vault_$words[1]
+        # print $curcontext
+    fi
+}
+
+compdef _vault vault
```

Comment:
@mcornella thanks for reviewing this :). I have renamed the file to `_vault` as you suggested. I have also removed the `_vault()` function and added `_vault "$@"` at the end of the file.

After I reload zsh if I write `vault` and hit `\t` I get the following error:

```bash
[...]
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
_tags:comptags:60: nesting level too deep
_tags:comptags:67: nesting level too deep
_tags:comptags:36: nesting level too deep
[1]    29100 segmentation fault  zsh
```

I also post here the last lines from the `_vault` file.

```bash

local curcontext=$curcontext ret=1
_arguments : \
    ${main_args[@]} \
    '*:: :->subcommands' && ret=0
if ((CURRENT == 1 )); then
    _vault_commands && ret=0
fi
if [[ $state == subcommands ]]; then
    # (( CURRENT -- ))
    curcontext="${curcontext%:*:*}:vault-$words[1]:"
    _call_function ret _vault_$words[1]
fi

_vault "$@"
```

What can I do next?